webapp/src/components/gitlab_project_selector/gitlab_project_selector.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Theme} from 'mattermost-redux/types/preferences';
+
+import ReactSelectSetting from '../react_select_setting';
+import {Project, Selection} from '../../types/gitlab_project_selector'
+
+interface PropTypes{
+    yourProjects: Project[];
+    theme: Theme;
+    onChange: (project: Selection) => void;
+    actions: any;
+    value: string;
+    addValidate: (key: string, validateField: () => boolean) => void;
+    removeValidate: (key: string) => void;
+};
+
+interface StateTypes{
+    invalid: boolean;
+    error: null|string;
+    isLoading: boolean;
+}
+
+const initialState = {
+    invalid: false,
+    error: null,
+    isLoading: false,
+};
+
+export default class GitlabProjectSelector extends PureComponent<PropTypes, StateTypes> {
+    
+    constructor(props: PropTypes) {
+        super(props);
```

Comment:
I think this call is now deprecated and there's no need for this.

Reply:
discussed